swtr/server.py

```diff
@@ -63,15 +79,15 @@ def index():
     #print auth_tok
     payload = {'what': 'img-anno',
                'access_token': auth_tok['access_token']}
-    req = requests.get(config.swtstoreURL + '/api/sweets/q', params=payload)
+    req = requests.get(current_app.config.get('SWTSTORE_URL', 'SWTSTORE_URL') + '/api/sweets/q', params=payload)
     sweets = req.json()
     return render_template('index.html', access_token=auth_tok['access_token'],
                            refresh_token=auth_tok['refresh_token'],
-                           config=config, url=request.args.get('where'),
-                           bookmark=quote_plus(config.app_url), sweets=sweets)
+                           config=current_app.config, url=request.args.get('where'),
+                           bookmark=quote_plus(current_app.config.APP_URL), sweets=sweets)
```

Comment:
does it load the config?

I'm using local_config.py (a copy of sample_config.py) -- note that I had
to change it to UPPERCASE because of Flask

and i'm seeing the forgotten config.app_url (should be
current_app.config.APP_URL)

btw: that weird error is probably related to url_for - I've added Flask
blueprint, so now it becomes url_for('swtr.annotate_webpage')

sorry for that, I would have tested it better had I known the config
values? (can I run the app locally with some guest access to swtr api?)

On Sun, Aug 23, 2015 at 4:07 AM, Anon Ray <notifications@github.com> wrote:

> In swtr/server.py
> <https://github.com/janastu/swtr/pull/44#discussion_r37703496>:
>
> >      sweets = req.json()
> >      return render_template('index.html', access_token=auth_tok['access_token'],
> >                             refresh_token=auth_tok['refresh_token'],
> > -                           config=config, url=request.args.get('where'),
> > -                           bookmark=quote_plus(config.app_url), sweets=sweets)
> > +                           config=current_app.config, url=request.args.get('where'),
> > +                           bookmark=quote_plus(current_app.config.APP_URL), sweets=sweets)
>
> @romanchyla <https://github.com/romanchyla> When I try to run this
> application, I get this error:
>
> "AttributeError: 'Config' object has no attribute 'APP_URL'"
>
> I believe we'd have to change all occurrences of
> current_app.config.SOME_KEY to current_app.config.get('SOME_KEY').
>
> When I tried to fix that, I keep getting BuildError: ('annotate_webpage',
> {}, None) which is weird. I don't understand why.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/janastu/swtr/pull/44/files#r37703496>.
>